pkg/pool-weighted/contracts/lbp/FixedPriceLBPool.sol

```diff
@@ -0,0 +1,336 @@
+        }
+
+        _projectTokenRate = projectTokenRate;
+    }
+
+    /**
+     * @notice Get the minimum allowed project token rate.
+     * @dev This guards against configuration errors, while retaining flexibility to accommodate normal use cases.
+     * @return minProjectTokenRate The minimum allowed token rate (i.e., price of the token in terms of the reserve)
+     */
+    function getMinimumProjectTokenRate() external pure returns (uint256) {
+        return _MIN_PROJECT_TOKEN_RATE;
+    }
+
+    /**
+     * @notice Get the maximum allowed project token rate.
+     * @dev This guards against configuration errors, while retaining flexibility to accommodate normal use cases.
+     * @return maxProjectTokenRate The maximum allowed token rate (i.e., price of the token in terms of the reserve)
+     */
+    function getMaximumProjectTokenRate() external pure returns (uint256) {
+        return _MAX_PROJECT_TOKEN_RATE;
+    }
+
+    /**
+     * @notice For bi-directional (buy and sell) sales, this is the maximum deviation from proportional initialization.
+     * @return initializationTolerancePercentage The maximum deviation from proportionality allowed at init time
+     */
+    function getInitializationTolerancePercentage() external pure returns (uint256) {
+        return _INITIALIZATION_TOLERANCE_PERCENTAGE;
+    }
+
+    /// @inheritdoc IFixedPriceLBPool
+    function getProjectTokenRate() external view returns (uint256) {
+        return _projectTokenRate;
+    }
+
+    /// @inheritdoc IFixedPriceLBPool
+    function getFixedPriceLBPoolDynamicData() external view returns (FixedPriceLBPoolDynamicData memory data) {
+        data.balancesLiveScaled18 = _vault.getCurrentLiveBalances(address(this));
+        data.staticSwapFeePercentage = _vault.getStaticSwapFeePercentage((address(this)));
+        data.totalSupply = totalSupply();
+
+        PoolConfig memory poolConfig = _vault.getPoolConfig(address(this));
+        data.isPoolInitialized = poolConfig.isPoolInitialized;
+        data.isPoolPaused = poolConfig.isPoolPaused;
+        data.isPoolInRecoveryMode = poolConfig.isPoolInRecoveryMode;
+        data.isSwapEnabled = _isSwapEnabled();
+    }
+
+    /// @inheritdoc IFixedPriceLBPool
+    function getFixedPriceLBPoolImmutableData() external view returns (FixedPriceLBPoolImmutableData memory data) {
+        data.tokens = _vault.getPoolTokens(address(this));
+        data.projectTokenIndex = _projectTokenIndex;
+        data.reserveTokenIndex = _reserveTokenIndex;
+
+        (data.decimalScalingFactors, ) = _vault.getPoolTokenRates(address(this));
+        data.isProjectTokenSwapInBlocked = _blockProjectTokenSwapsIn;
+        data.startTime = _startTime;
+        data.endTime = _endTime;
+
+        data.projectTokenRate = _projectTokenRate;
+
+        // Migration-related params, non-zero if the pool supports migration.
+        data.migrationRouter = _migrationRouter;
```

Comment:
The migration router computes the spot price of the LBP, but that assumes it's a weighted pool. I'd consider not allowing this for now.